data/extension/init/templates/tinygo/envoy.filters.http/default/main_test.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxytest"
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/types"
+)
+
+func TestHttpHeaders_OnHttpRequestHeaders(t *testing.T) {
+	opt := proxytest.NewEmulatorOption().
+		WithNewRootContext(newRootContext).
+		WithNewHttpContext(newHttpContext)
+	host := proxytest.NewHostEmulator(opt)
+	defer host.Done() // release the host emulation lock so that other test cases can insert their own host emulation
+
+	host.StartVM() // call OnVMStart -> the metric is initialized
+
+	contextID := host.HttpFilterInitContext() // create http stream
+
+	hs := [][2]string{
+		{"key1", "value1"},
+		{"key2", "value2"},
+	}
+
+	host.HttpFilterPutRequestHeaders(contextID, hs) // call OnHttpRequestHeaders
```

Comment:
[nit] I think `HttpFilterPutRequestHeaders` is not an intuitive name

Reply:
all the filter specific emulation is prefixed by filter types e.g. NetworkFilter, HttpFilter: https://github.com/tetratelabs/proxy-wasm-go-sdk/blob/main/proxytest/proxytest.go#L42-L68

Reply:
It's `Put` in the name that confuses me.
* "Put" where ?
* Or "Put" like HTTP verb "PUT" ?

I wouldn't be able to understand what this method does without a comment `// call OnHttpRequestHeaders`.

Usually, it's a sign that method name could be improved. E.g., `CallOnHttpRequestHeaders`.

BTW, why does `host.CallOnLogForAccessLogger` break this naming convention ? Why not `host.AccessLoggerPutLog` then ?

Reply:
OK, now I see why you are confused. This could be improved.